ehrql/dummy_data/query_info.py

```diff
@@ -41,15 +40,11 @@ def __post_init__(self):
     def record_value(self, value):
         if hasattr(value, "_to_primitive_type"):
             value = value._to_primitive_type()
-        self.values_used.add(value)
+        self._values_used.add(value)
 
     @cached_property
-    def choices(self):
-        cat_constraint = self.get_constraint(Constraint.Categorical)
-        if cat_constraint is not None:
-            return cat_constraint.values
-        else:
-            return sorted(self.values_used)
+    def values_used(self):
+        return set(sorted(self._values_used))
```

Comment:
I don't think you want the `set()` wrapper here: `_values_used` is already a set, so this is just undoing the affect of the sorting which we want to give us deterministic dummy data.

Reply:
I added this to ensure the return type was consistent, but on reflection I was wrong: whilst `_values_used` is a set, `values_used` need not be. I'd also forgotten that `set`s don't maintain insertion order. Thanks for your comment!